src/v/cluster/tests/rm_stm_tests.cc

```diff
@@ -931,3 +931,95 @@ FIXTURE_TEST(test_tx_expiration_without_data_batches, rm_stm_test_fixture) {
                != expired.end();
     }).get();
 }
+
+/*
+ * This test ensures concurrent evictions can happen in the presence of
+ * replication operations and operations that reset the state (snapshots,
+ * partition stop).
+ */
+FIXTURE_TEST(test_concurrent_producer_evictions, rm_stm_test_fixture) {
```

Comment:
can you enable stress fiber in this test ? From my experience it is great in uncovering all sorts of race conditions